### ISSUE TITLE
Bunches not respecting auto-open/close days and 

### ISSUE THREAD

**Issue by Stever888** [ttscoff/bunch#223]
I love love love love love Bunch.

Recently I've been having some problems with Bunches not opening and closing at the scheduled times. This was working great until a few weeks ago and now seems broken. I don't know if an upgrade changed how scheduling works, or whether my installation is corrupted, or whether my frontmatter is screwed up.

The bunch with this headmatter, for example, is only supposed to open on weekdays, but it opens on weekends as well:

```
---
open on: MTWThF 06:25 am
close at: 8:30 am
# only auto-open on my work machine
schedule if: <<uuid>>

#menu divider: before
menu order: 5
---
```

The bunch with this frontmatter opens fine, and it even sends a notification that it's going to close in a few minute ... but it doesn't actually close.

```
---
open on: MTWThF 08:41
close at: 18:28
ignores state: true
# only auto-open on my work machine
schedule if: 0892F16F-2118-56E6-9274-140DE8705051
menu order: 1
---
```

**Comment by ttscoff** [comment]
Sorry I took so long to reply to this one. The scheduling stuff is hard to test, still trying to figure out a way to replicate some of these weirder issues. I don't see anything wrong with your frontmatter, so I'll need to just keep trying to replicate.

**Comment by Stever888** [comment]
Hi, Brett,

Thanks for getting back to me. Isn't debugging real time software *fun*? (I suddenly furiously start remembering why I left software.) I'm still having problems with open/close sometimes not happening, or happening on the wrong days, etc. Is there any way I can tell Bunch to nuke its thoughts of scheduling and whatever cron jobs (or whatever the mechanism is) that it has scheduled, and rebuild its schedule from scratch?

Curiously,

Stever
(he/his)

-- 
iTunes top-10 business podcast (2007-2020): getitdoneguy.com/archive
Start finishing with Get-it-Done Groups: www.getitdonegroups.com
Executive coaching: SteverRobbins.com
+1 (617) 354-1446

NEED TO REACH ME QUICKLY? Text 617-354-1446 xxx Email won’t do it. I only check once a day.

**Comment by ttscoff** [comment]
The schedule gets built every time it loads a folder. It parses 
everything with a schedule, figures out what the next unscheduled time 
is, and adds an NSTimer to trigger at that time. When an event occurs, 
the schedules should be reparsed to determine the next event. So really, 
you shouldn't need to nuke anything, and quitting and restarting is *for 
sure* going to rebuild the schedule from scratch.

-Brett

**Comment by ttscoff** [comment]
Yeah, scheduling seems to go wacky for some folks and not others. 
Because I haven't been able to replicate problems like this on my own 
machines, I've had a bear of a time tracking down the culprit. Will test 
your specific issues locally and see if I can replicate.

FYI no changes have been made to scheduling in the last few versions, so 
any issues that suddenly manifested would indicate some other change in 
the system. What that would be I have no idea yet.

-Brett

**Comment by Chris-May** [comment]
I can't imagine this will help much, but I have a similar issue, and the log shows how a bunch is scheduled for a specific time, yet it launches earlier.

For example, I have a shutdown reminder task to open at 5pm, and it opened today at 4:34pm.
```
2023-07-28 16:34:55: 🟡[awake/shutdown] Scheduled to open weekly starting 7/28/23, 5:00 PM
```

There might be some time drift if I leave my computer (and therefore bunches) up for days.
